TestStack.Dossier.Tests/BuilderBuildTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Shouldly;
+using TestStack.Dossier.Tests.Builders;
+using TestStack.Dossier.Tests.Entities;
+using Xunit;
+
+namespace TestStack.Dossier.Tests
+{
+    public class BuilderBuildTests
+    {
+        [Fact]
+        public void GivenBasicBuilder_WhenCallingBuildExplicitly_ThenReturnAnObject()
+        {
+            var builder = Builder<Customer>.CreateNew();
+
+            var customer = builder.Build();
+
+            customer.ShouldBeOfType<Customer>();
+        }
+
+        [Fact]
+        public void GivenBuilder_WhenCallingSetExplicitly_ShouldOverrideValues()
+        {
+            var builder = Builder<Customer>.CreateNew()
```

Comment:
I think if that is a concern, then the consumer will override the BuildObject method and construct it with the appropriate constructor, using the original behaviour? This is potentially a benefit for view models and DTOs.